cirq-core/cirq/sim/act_on_args.py

```diff
@@ -47,6 +47,7 @@ def __init__(
         prng: np.random.RandomState = None,
         qubits: Sequence['cirq.Qid'] = None,
         log_of_measurement_results: Dict[str, List[int]] = None,
+        measured_qubits: Dict[str, Tuple['cirq.Qid', ...]] = None,
```

Comment:
@95-martin-orion I did this the least-invasive way, of adding this member and propagating it everywhere it needed to be. This is a little smelly because it and `log_of_measurement_results` go hand in hand and are initialized, copied, etc. in pairs. I think the better option though is to introduce a new `classical_state: ClassicalState` that contains both of these, plus whatever we add later, and then deprecate `log_of_measurement_results`.. I know it breaks the "rule of three" to do so, but I'm pretty sure we'll have a third thing to put into it soon.

It may be a big deprecation, but it may be worth it to do now rather than having to clean up the mess *after* things like multiple measurements, [keeping track of qubits](https://github.com/quantumlib/Cirq/issues/4449), etc have all become dependent on it.

Additionally I noticed a minor bug here is that I don't keep track of the qubits in keyed channels. If we have a ClassicalState class then we can make the interface such that writing to the dictionaries requires both a value and a qubit list. Of course, that will make for a much bigger review. WDYT?

Reply:
Hmm, per the bug of not keeping track of qubits in keyed channels, I notice there that the type of the measurement is different anyway. For regular measurements, it's List[int] with an int per qubit. But for these channel measurements, it's just a single int to represent the whole measurement.

If we create a ClassicalState class, I'd propose putting those into a separate field. WDYT?

Reply:
Actually maybe it's better to keep them (measurements and channel measurements) together so they can both be used in classical control conditions? I'd just have to update the resolver to check the type: if it's `int` then straight return it, otherwise do the big-endian conversion.

Note there's a pre-existing bug here: https://github.com/quantumlib/Cirq/blob/01ae51eebf3b18a5cbee9fc0c697d4e1511c07f2/cirq-core/cirq/study/result.py#L123: `Result..data` assumes measurements are all `List[List[int]]`, as stated in the `Result.__init__` docstring, so that would break if there's a channel measurement. (It also produces invalid data for qudits).

Feeling more and more like we need this ClassicalState class to encapsulate these things.